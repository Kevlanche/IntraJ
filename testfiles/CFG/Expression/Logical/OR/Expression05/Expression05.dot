digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[6]"; "Opt[9]"; "List[10]"; "List[11]"}
{ rank = same; "List[7]"; "MethodDecl[12]"}
{ rank = same; "Modifier[8]"; "Entry[13]"; "Modifiers[14]"; "PrimitiveTypeAccess[16]"; "List[17]"; "List[26]"; "Opt[27]"; "Exit[28]"}
{ rank = same; "List[15]"; "ParameterDeclaration[18]"; "ParameterDeclaration[22]"; "Block[28]"}
{ rank = same; "Modifiers[19]"; "PrimitiveTypeAccess[21]"; "Modifiers[23]"; "PrimitiveTypeAccess[25]"; "List[29]"}
{ rank = same; "List[20]"; "List[24]"; "ForStmt[30]"; "ExprStmt[59]"}
{ rank = same; "List[31]"; "Opt[41]"; "List[49]"; "Block[53]"; "AssignSimpleExpr[60]"}
{ rank = same; "VarDeclStmt[32]"; "OrLogicalExpr[42]"; "ExprStmt[50]"; "List[54]"; "VarAccess[61]"; "IntegerLiteral[62]"}
{ rank = same; "Modifiers[33]"; "PrimitiveTypeAccess[35]"; "List[36]"; "GTExpr[43]"; "LTExpr[46]"; "PostIncExpr[51]"; "ExprStmt[55]"}
{ rank = same; "List[34]"; "VariableDeclarator[37]"; "VarAccess[44]"; "IntegerLiteral[45]"; "VarAccess[47]"; "IntegerLiteral[48]"; "VarAccess[52]"; "AssignSimpleExpr[56]"}
{ rank = same; "List[38]"; "Opt[39]"; "VarAccess[57]"; "IntegerLiteral[58]"}
{ rank = same; "IntegerLiteral[40]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < Expression05 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[6]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[7]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[8]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[9]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[10]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[11]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[12]" [label="MethodDecl\n < m >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[13]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[14]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[15]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[16]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[17]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[18]" [label="ParameterDeclaration\n < a >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[19]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[20]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[21]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[22]" [label="ParameterDeclaration\n < b >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[23]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[24]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[25]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[27]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[28]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[29]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ForStmt[30]" [label="ForStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[31]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[32]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[33]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[34]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[35]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[36]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[37]" [label="VariableDeclarator\n < int c >", fillcolor=white   style=filled  ]
"List[38]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[39]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[40]" [label="IntegerLiteral\n < 0 >", fillcolor=white   style=filled  ]
"Opt[41]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"OrLogicalExpr[42]" [label="OrLogicalExpr\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"GTExpr[43]" [label="GTExpr\n ", fillcolor=white   style=filled  ]
"VarAccess[44]" [label="VarAccess\n < a >", fillcolor=white   style=filled  ]
"IntegerLiteral[45]" [label="IntegerLiteral\n < 0 >", fillcolor=white   style=filled  ]
"LTExpr[46]" [label="LTExpr\n ", fillcolor=white   style=filled  ]
"VarAccess[47]" [label="VarAccess\n < a >", fillcolor=white   style=filled  ]
"IntegerLiteral[48]" [label="IntegerLiteral\n < 10 >", fillcolor=white   style=filled  ]
"List[49]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[50]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PostIncExpr[51]" [label="PostIncExpr\n < a >", fillcolor=white   style=filled  ]
"VarAccess[52]" [label="VarAccess\n < a >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[53]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[54]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[55]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"AssignSimpleExpr[56]" [label="AssignSimpleExpr\n < c >", fillcolor=white   style=filled  ]
"VarAccess[57]" [label="VarAccess\n < c >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[58]" [label="IntegerLiteral\n < 1 >", fillcolor=white   style=filled  ]
"ExprStmt[59]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"AssignSimpleExpr[60]" [label="AssignSimpleExpr\n < a >", fillcolor=white   style=filled  ]
"VarAccess[61]" [label="VarAccess\n < a >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[62]" [label="IntegerLiteral\n < 2 >", fillcolor=white   style=filled  ]
"Exit[28]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[7]" -> "Modifier[8]" [style=dashed, color=gray]
"Modifiers[6]" -> "List[7]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[9]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[10]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Entry[13]" [style=dashed, color=gray]
"Modifiers[14]" -> "List[15]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Modifiers[14]" [style=dashed, color=gray]
"MethodDecl[12]" -> "PrimitiveTypeAccess[16]" [style=dashed, color=gray]
"Modifiers[19]" -> "List[20]" [style=dashed, color=gray]
"ParameterDeclaration[18]" -> "Modifiers[19]" [style=dashed, color=gray]
"ParameterDeclaration[18]" -> "PrimitiveTypeAccess[21]" [style=dashed, color=gray]
"List[17]" -> "ParameterDeclaration[18]" [style=dashed, color=gray]
"Modifiers[23]" -> "List[24]" [style=dashed, color=gray]
"ParameterDeclaration[22]" -> "Modifiers[23]" [style=dashed, color=gray]
"ParameterDeclaration[22]" -> "PrimitiveTypeAccess[25]" [style=dashed, color=gray]
"List[17]" -> "ParameterDeclaration[22]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[17]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[26]" [style=dashed, color=gray]
"Modifiers[33]" -> "List[34]" [style=dashed, color=gray]
"VarDeclStmt[32]" -> "Modifiers[33]" [style=dashed, color=gray]
"VarDeclStmt[32]" -> "PrimitiveTypeAccess[35]" [style=dashed, color=gray]
"VariableDeclarator[37]" -> "List[38]" [style=dashed, color=gray]
"Opt[39]" -> "IntegerLiteral[40]" [style=dashed, color=gray]
"VariableDeclarator[37]" -> "Opt[39]" [style=dashed, color=gray]
"List[36]" -> "VariableDeclarator[37]" [style=dashed, color=gray]
"VarDeclStmt[32]" -> "List[36]" [style=dashed, color=gray]
"List[31]" -> "VarDeclStmt[32]" [style=dashed, color=gray]
"ForStmt[30]" -> "List[31]" [style=dashed, color=gray]
"GTExpr[43]" -> "VarAccess[44]" [style=dashed, color=gray]
"GTExpr[43]" -> "IntegerLiteral[45]" [style=dashed, color=gray]
"OrLogicalExpr[42]" -> "GTExpr[43]" [style=dashed, color=gray]
"LTExpr[46]" -> "VarAccess[47]" [style=dashed, color=gray]
"LTExpr[46]" -> "IntegerLiteral[48]" [style=dashed, color=gray]
"OrLogicalExpr[42]" -> "LTExpr[46]" [style=dashed, color=gray]
"Opt[41]" -> "OrLogicalExpr[42]" [style=dashed, color=gray]
"ForStmt[30]" -> "Opt[41]" [style=dashed, color=gray]
"PostIncExpr[51]" -> "VarAccess[52]" [style=dashed, color=gray]
"ExprStmt[50]" -> "PostIncExpr[51]" [style=dashed, color=gray]
"List[49]" -> "ExprStmt[50]" [style=dashed, color=gray]
"ForStmt[30]" -> "List[49]" [style=dashed, color=gray]
"AssignSimpleExpr[56]" -> "VarAccess[57]" [style=dashed, color=gray]
"AssignSimpleExpr[56]" -> "IntegerLiteral[58]" [style=dashed, color=gray]
"ExprStmt[55]" -> "AssignSimpleExpr[56]" [style=dashed, color=gray]
"List[54]" -> "ExprStmt[55]" [style=dashed, color=gray]
"Block[53]" -> "List[54]" [style=dashed, color=gray]
"ForStmt[30]" -> "Block[53]" [style=dashed, color=gray]
"List[29]" -> "ForStmt[30]" [style=dashed, color=gray]
"AssignSimpleExpr[60]" -> "VarAccess[61]" [style=dashed, color=gray]
"AssignSimpleExpr[60]" -> "IntegerLiteral[62]" [style=dashed, color=gray]
"ExprStmt[59]" -> "AssignSimpleExpr[60]" [style=dashed, color=gray]
"List[29]" -> "ExprStmt[59]" [style=dashed, color=gray]
"Block[28]" -> "List[29]" [style=dashed, color=gray]
"Opt[27]" -> "Block[28]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Opt[27]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Exit[28]" [style=dashed, color=gray]
"List[11]" -> "MethodDecl[12]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[11]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"PostIncExpr[51]" -> "VarAccess[44]" [color=blue, constraint=false, xlabel=" "] 
"PostIncExpr[51]" -> "AssignSimpleExpr[56]" [color=red, constraint=false]
"AssignSimpleExpr[56]" -> "PostIncExpr[51]" [color=blue, constraint=false, xlabel=" "] 
"AssignSimpleExpr[56]" -> "IntegerLiteral[58]" [color=red, constraint=false]
"IntegerLiteral[58]" -> "AssignSimpleExpr[56]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[58]" -> "GTExpr[43]" [color=red, constraint=false]
"IntegerLiteral[58]" -> "LTExpr[46]" [color=red, constraint=false]
"Exit[28]" -> "AssignSimpleExpr[60]" [color=red, constraint=false]
"AssignSimpleExpr[60]" -> "Exit[28]" [color=blue, constraint=false, xlabel=" "] 
"AssignSimpleExpr[60]" -> "IntegerLiteral[62]" [color=red, constraint=false]
"IntegerLiteral[62]" -> "AssignSimpleExpr[60]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[62]" -> "LTExpr[46]" [color=red, constraint=false]
"LTExpr[46]" -> "IntegerLiteral[58]" [color=blue, constraint=false, xlabel="T "] 
"LTExpr[46]" -> "IntegerLiteral[62]" [color=blue, constraint=false, xlabel="F "] 
"LTExpr[46]" -> "IntegerLiteral[48]" [color=red, constraint=false]
"IntegerLiteral[48]" -> "LTExpr[46]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[48]" -> "VarAccess[47]" [color=red, constraint=false]
"VarAccess[47]" -> "IntegerLiteral[48]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[47]" -> "GTExpr[43]" [color=red, constraint=false]
"GTExpr[43]" -> "IntegerLiteral[58]" [color=blue, constraint=false, xlabel="T "] 
"GTExpr[43]" -> "VarAccess[47]" [color=blue, constraint=false, xlabel="F "] 
"GTExpr[43]" -> "IntegerLiteral[45]" [color=red, constraint=false]
"IntegerLiteral[45]" -> "GTExpr[43]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[45]" -> "VarAccess[44]" [color=red, constraint=false]
"VarAccess[44]" -> "IntegerLiteral[45]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[44]" -> "VariableDeclarator[37]" [color=red, constraint=false]
"VarAccess[44]" -> "PostIncExpr[51]" [color=red, constraint=false]
"VariableDeclarator[37]" -> "VarAccess[44]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[37]" -> "IntegerLiteral[40]" [color=red, constraint=false]
"IntegerLiteral[40]" -> "VariableDeclarator[37]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[40]" -> "Entry[13]" [color=red, constraint=false]
"Entry[13]" -> "IntegerLiteral[40]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"Modifiers[6]" -> "Opt[9]" -> "List[10]" -> "List[11]" [style=invis]
"Entry[13]" -> "Modifiers[14]" -> "PrimitiveTypeAccess[16]" -> "List[17]" -> "List[26]" -> "Opt[27]" -> "Exit[28]" [style=invis]
"ParameterDeclaration[18]" -> "ParameterDeclaration[22]" [style=invis]
"Modifiers[19]" -> "PrimitiveTypeAccess[21]" [style=invis]
"Modifiers[23]" -> "PrimitiveTypeAccess[25]" [style=invis]
"ForStmt[30]" -> "ExprStmt[59]" [style=invis]
"List[31]" -> "Opt[41]" -> "List[49]" -> "Block[53]" [style=invis]
"Modifiers[33]" -> "PrimitiveTypeAccess[35]" -> "List[36]" [style=invis]
"List[38]" -> "Opt[39]" [style=invis]
"GTExpr[43]" -> "LTExpr[46]" [style=invis]
"VarAccess[44]" -> "IntegerLiteral[45]" [style=invis]
"VarAccess[47]" -> "IntegerLiteral[48]" [style=invis]
"VarAccess[57]" -> "IntegerLiteral[58]" [style=invis]
"VarAccess[61]" -> "IntegerLiteral[62]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
64", fillcolor=1]
Edges [label="Total Nr. Edges
95", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
15", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
16", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}