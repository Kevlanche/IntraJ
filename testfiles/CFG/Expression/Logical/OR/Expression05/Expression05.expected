NodeName: Entry[13] Successors: [IntegerLiteral[40]]
NodeName: IntegerLiteral[40] Successors: [VariableDeclarator[37]]
NodeName: VariableDeclarator[37] Successors: [VarAccess[44]]
NodeName: VarAccess[44] Successors: [IntegerLiteral[45]]
NodeName: IntegerLiteral[45] Successors: [GTExpr[43]]
NodeName: GTExpr[43] Successors: [IntegerLiteral[58], VarAccess[47]]
NodeName: IntegerLiteral[58] Successors: [AssignSimpleExpr[56]]
NodeName: AssignSimpleExpr[56] Successors: [PostIncExpr[51]]
NodeName: PostIncExpr[51] Successors: [VarAccess[44]]
NodeName: VarAccess[47] Successors: [IntegerLiteral[48]]
NodeName: IntegerLiteral[48] Successors: [LTExpr[46]]
NodeName: LTExpr[46] Successors: [IntegerLiteral[58], IntegerLiteral[62]]
NodeName: IntegerLiteral[62] Successors: [AssignSimpleExpr[60]]
NodeName: AssignSimpleExpr[60] Successors: [Exit[28]]
NodeName: Exit[28] Successors: []

NodeName: Exit[28] Predecessor: [AssignSimpleExpr[60]]
NodeName: AssignSimpleExpr[60] Predecessor: [IntegerLiteral[62]]
NodeName: IntegerLiteral[62] Predecessor: [LTExpr[46]]
NodeName: LTExpr[46] Predecessor: [IntegerLiteral[48]]
NodeName: IntegerLiteral[48] Predecessor: [VarAccess[47]]
NodeName: VarAccess[47] Predecessor: [GTExpr[43]]
NodeName: GTExpr[43] Predecessor: [IntegerLiteral[45]]
NodeName: IntegerLiteral[45] Predecessor: [VarAccess[44]]
NodeName: VarAccess[44] Predecessor: [VariableDeclarator[37], PostIncExpr[51]]
NodeName: VariableDeclarator[37] Predecessor: [IntegerLiteral[40]]
NodeName: IntegerLiteral[40] Predecessor: [Entry[13]]
NodeName: Entry[13] Predecessor: []
NodeName: PostIncExpr[51] Predecessor: [AssignSimpleExpr[56]]
NodeName: AssignSimpleExpr[56] Predecessor: [IntegerLiteral[58]]
NodeName: IntegerLiteral[58] Predecessor: [GTExpr[43], LTExpr[46]]
